4.analyze_data/notebooks/UMAP/nbconverted/1.visualize-UMAP-plate3.r

```diff
@@ -319,3 +393,137 @@ merged_plot <- ggplot(combined_df, aes(x = UMAP0, y = UMAP1, color = Group)) +
 print(merged_plot)
 ggsave(output_file, merged_plot, dpi = 500, height = 12, width = 12)
 
+
+# Remove decimal from nuclei object number to avoid issues with merging
+umap_cp_df <- umap_cp_df %>%
+  mutate(Metadata_Nuclei_Number_Object_Number = gsub("\\.0", "", as.character(Metadata_Nuclei_Number_Object_Number)))
+
+# Load in parquet file with nearest neighbors per single cell (not normalized)
+neighbors_df <- arrow::read_parquet(
+    "../../../3.process_cfret_features/data/single_cell_profiles/localhost230405150001_sc_annotated.parquet"
+)
+
+desired_columns <- c("Cells_Neighbors_NumberOfNeighbors_Adjacent", 
+                     "Metadata_Well", 
+                     "Metadata_Site", 
+                     "Metadata_Nuclei_Number_Object_Number")
+
+neighbors_df <- neighbors_df[, desired_columns]
+
+neighbors_df <- neighbors_df %>%
+  mutate(Metadata_Nuclei_Number_Object_Number = as.character(Metadata_Nuclei_Number_Object_Number))
+
+# Assuming you want to merge based on the specified columns
+merge_columns <- c("Metadata_Well", "Metadata_Site", "Metadata_Nuclei_Number_Object_Number")
+
+# Merge number of neighbors based on cell to umap dataframe
+merged_data <- merge(neighbors_df, umap_cp_df, by = merge_columns)
+
+umap_cell_neighbors_figure <- 
+  ggplot(merged_data, aes(x = UMAP0, y = UMAP1)) +
+  geom_point(aes(color = Cells_Neighbors_NumberOfNeighbors_Adjacent), alpha = 0.5) +
+  theme_bw() +
+  scale_color_continuous(
+    name = "Number of Cell Neighbors", 
+    low = "light blue",
+    high = "blue"
+  ) +
+  labs(title = "UMAP of Plate 3 Morphology Space Comparing\nBetween Number of Neighbors per Cell", x = "UMAP0", y = "UMAP1") +
+  theme(
+    # Increase title size
+    plot.title = element_text(size = 20),
+    
+    # Increase axis text size
+    axis.title = element_text(size = 14),
+    axis.text = element_text(size = 12),
+    
+    # Increase legend size
+    legend.title = element_text(size = 14),
+    legend.text = element_text(size = 14),
+  ) +
+  coord_fixed(1.1)
```

Comment:
For some of these theme items, you could set up an R theme file and import it to reduce code lines. 
An example of one of those files: https://github.com/WayScience/phenotypic_profiling/blob/main/7.figures/themes.r

and the implementation: https://github.com/WayScience/phenotypic_profiling/blob/main/7.figures/f1_score_visualization.ipynb

Reply:
I will include this as an issue to address for a full refactor later on. I would like this file to be for all the plots, not just the few we will use for the manuscript. Thanks for the suggestion!